app/TextParser/ProductsTableRelatedModule.php

```diff
@@ -31,75 +30,78 @@ class ProductsTableRelatedModule extends Base
 	public function process()
 	{
 		$html = '';
-		$relatedModuleRecordId = $this->textParser->recordModel->get($this->relatedModulesFields[$this->textParser->moduleName]);
-		$relatedModuleRecordModel = \Vtiger_Record_Model::getInstanceById($relatedModuleRecordId);
-		if (!$relatedModuleRecordModel->getModule()->isInventory()) {
-			return $html;
-		}
-		$relatedModuleName = $relatedModuleRecordModel->getModuleName();
-		$inventory = \Vtiger_Inventory_Model::getInstance($relatedModuleName);
-		$inventoryRows = $relatedModuleRecordModel->getInventoryData();
-		$headerStyle = 'font-size:9px;padding:0px 4px;';
-		$bodyStyle = 'font-size:8px;border:1px solid #ddd;padding:0px 4px;';
-		$html .= '<table class="products-table-related-module" style="width:100%;border-collapse:collapse;"><thead><tr>';
-		$groupModels = [];
-		foreach (['ItemNumber', 'Name', 'Quantity', 'Discount', 'Currency', 'DiscountMode', 'TaxMode', 'UnitPrice', 'GrossPrice', 'NetPrice', 'Tax', 'TotalPrice', 'Value'] as $fieldType) {
-			foreach ($inventory->getFieldsByType($fieldType) as $fieldModel) {
-				if (!$fieldModel->isVisible()) {
+		$givenRelatedModuleName = $this->params[0];
+		$relatedModuleField = $this->params[1];
+		$fieldsToShow = !empty($this->params[2]) ? explode(',', $this->params[2]) : ['seq', 'name', 'qty', 'discount', 'currency', 'discountmode', 'taxmode', 'price', 'gross', 'net', 'tax', 'total'];
```

Comment:
[$relatedModuleName, $relatedModuleField , $fieldsToShow] = array_pad($this->params, 3, null);